test/balancer_test.go

```diff
@@ -154,6 +154,7 @@ func (s) TestCredsBundleFromBalancer(t *testing.T) {
 	te.tapHandle = authHandle
 	te.customDialOptions = []grpc.DialOption{
 		grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":{}}]}`, testBalancerName)),
+		grpc.WithAuthority("x.test.example.com"),
```

Comment:
If I understand correctly, the changes to the following tests mean that the `CredsBundle` provided by the balancer though [NewSubConnOpts](https://github.com/grpc/grpc-go/blob/62ec29fd9b3f9ea3cea6dc08a31e837aa92678b7/balancer/balancer.go#L100-L106) or the [DialOption](https://github.com/grpc/grpc-go/blob/62ec29fd9b3f9ea3cea6dc08a31e837aa92678b7/dialoptions.go#L447-L451) can no longer set the SNI for TLS. Is this change in behaviour expected?